lib/rucio/web/rest/flaskapi/v1/common.py

```diff
@@ -91,6 +91,17 @@ def dispatch_request(self, *args, **kwargs):
             return super(ErrorHandlingMethodView, self).dispatch_request(*args, **kwargs)
         except HTTPException:
             raise
+        except DatabaseException as error:
+            msg = f'DatabaseException in {self.__class__.__module__} {self.__class__.__name__} {flask.request.method}'
+            # logged, because this should be the __exception__
+            logging.debug(msg, exc_info=True)
+            return generate_http_error_flask(
+                status_code=503,
+                exc=error.__class__.__name__,
+                exc_msg='The Rucio database is overloaded. Please try again in a few minutes.',
```

Comment:
Let's be a bit more specific here.
Check for the `QueuePool` in the database exception and only then raise the 503. Otherwise we are raising exceptions which are not necessarily load related. I would also change the `exc_msg` to something more generic. `A database exception occurred. Please try again in a moment` or something like that.

Reply:
While I agree on the first, the message for QueuePool problem should mention that it's overloaded. Maybe something like "There are currently too many requests for the Rucio servers to handle. Please try again in a few minutes."